### PR TITLE
77648:   participant view in wizard

### PR DESCRIPTION
**Pull request by bratli**

### CODE REVIEW

File: src/modules/InvitationForPunchOut/components/RoleSelector/index.tsx [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    }, containerRef);
+
+    const toggleDropdown = (): void => {
+        setIsOpen(!isOpen);
+    };
+
+    const clearOtherRoles = (roleIndex: number): void => {
+        setAllRoles(ar => {
+            const copyR = [...ar];
+            copyR.forEach((r, i) => {
```
Comment (paal123) [inline src/modules/InvitationForPunchOut/components/RoleSelector/index.tsx:84]: Shold  we rather use more readable names? ar, copyR and r makes it a little  hard understand exactly what we are juggling here

File: src/modules/InvitationForPunchOut/views/CreateIPO/CreateIPO.tsx [hunk 2]
```diff
@@ -178,7 +191,7 @@ const CreateIPO = (): JSX.Element => {
                 previous={goToPreviousStep}
                 participants={participants}
                 setParticipants={setParticipants}
-                isValid={true}
+                isValid={steps[2].isCompleted}
```
Comment (paal123) [inline src/modules/InvitationForPunchOut/views/CreateIPO/CreateIPO.tsx:194]: Could we create an Enum for steps to be able to use more readable names? Like Steps.Participants and a then at the same time be able to do if (step > 2) or if (step>Steps.Participants)

File: src/modules/InvitationForPunchOut/views/CreateIPO/Participants/Participants.tsx [hunk 3]
```diff
@@ -1,36 +1,360 @@
...
+            return (): void => requestCanceler && requestCanceler();
+        } catch (error) {
+            showSnackbarNotification(error.message);
+        }
+    }, []);
+
+    useEffect(() => {
+        if(personFilter != '') {
+            try {
+                setFilteredPersons(testPart.filter(r => r.text.toLocaleLowerCase().startsWith(personFilter.toLocaleLowerCase())));
```
Comment (paal123) [inline src/modules/InvitationForPunchOut/views/CreateIPO/Participants/Participants.tsx:118]: testPart?

File: src/modules/InvitationForPunchOut/views/CreateIPO/Participants/Participants.tsx [hunk 4]
```diff
@@ -1,36 +1,360 @@
...
+        return JSON.parse(JSON.stringify(availableRoles));
+    };
+
+    const setOrganization = (value: string, index: number): void => {
+        setParticipants(p => {
+            const participantsCopy = [...p];
+            participantsCopy[index].organization = value;
+            return participantsCopy;
+        });
+        if(value == Organizations[6].value) {
```
Comment (paal123) [inline src/modules/InvitationForPunchOut/views/CreateIPO/Participants/Participants.tsx:137]: Organizations as enum?

**Review by paal123 — APPROVED** [review]
With suggestions

File: src/modules/InvitationForPunchOut/views/CreateIPO/Participants/Participants.tsx (hunk 3 above)
Comment (bratli) [reply src/modules/InvitationForPunchOut/views/CreateIPO/Participants/Participants.tsx:118]: just using this until endpoint are ready - it is just dummy data. I have created a PBI to get persons from endpoint when this is ready

**Pull request merged**